elephant/statistics.py

```diff
@@ -240,3 +243,283 @@ def lv(v):
+        TRI (triangle), GAU (gaussian), EPA (epanechnikov), EXP (exponential),
+        ALP (alpha function). EXP and ALP are asymmetric kernel forms and
+        assume optional parameter `direction`.
+    sigma : Quantity
+        Standard deviation of the distribution associated with kernel shape.
+        This parameter defines the time resolution of the kernel estimate
+        and makes different kernels comparable (cf. [1] for symmetric kernels).
+        This is used here as an alternative definition to the cut-off
+        frequency of the associated linear filter.
+    sampling_period : float
+        Temporal resolution of input and output.
+    direction : {-1, 1}
+        Asymmetric kernels have two possible directions.
+        The values are -1 or 1, default is 1. The
+        definition here is that for direction = 1 the
+        kernel represents the impulse response function
+        of the linear filter. Default value is 1.
+
+    Returns
+    -------
+    kernel : numpy.ndarray
+        Array of kernel. The length of this array is always an odd
+        number to represent symmetric kernels such that the center bin
+        coincides with the median of the numeric array, i.e for a
+        triangle, the maximum will be at the center bin with equal
+        number of bins to the right and to the left.
+   norm : float
+        For rate estimates. The kernel vector is normalized such that
+        the sum of all entries equals unity sum(kernel)=1. When
+        estimating rate functions from discrete spike data (0/1) the
+        additional parameter `norm` allows for the normalization to
+        rate in spikes per second.
+
+        For example:
+        ``rate = norm * scipy.signal.lfilter(kernel, 1, spike_data)``
+    m_idx : int
+        Index of the numerically determined median (center of gravity)
+        of the kernel function.
+
+    Examples
+    --------
+    To obtain single trial rate function of trial one should use::
+
+        r = norm * scipy.signal.fftconvolve(sua, kernel)
+
+    To obtain trial-averaged spike train one should use::
+
+        r_avg = norm * scipy.signal.fftconvolve(sua, np.mean(X,1))
+
+    where `X` is an array of shape `(l,n)`, `n` is the number of trials and
+    `l` is the length of each trial.
+
+    See also
+    --------
+    SpikeTrain.instantaneous_rate
+    SpikeList.averaged_instantaneous_rate
+
+    References
+    ----------
+    .. [1] Meier R, Egert U, Aertsen A, Nawrot MP, "FIND - a unified framework
+       for neural data analysis"; Neural Netw. 2008 Oct; 21(8):1085-93.
+
+    .. [2] Nawrot M, Aertsen A, Rotter S, "Single-trial estimation of neuronal
+       firing rates - from single neuron spike trains to population activity";
+       J. Neurosci Meth 94: 81-92; 1999.
+
+    """
+    forms_abbreviated = np.array(['BOX', 'TRI', 'GAU', 'EPA', 'EXP', 'ALP'])
+    forms_verbose = np.array(['rectangle', 'triangle', 'gaussian', 'epanechnikov',
```

Comment:
If you're going to keep the abbreviated names, then probably the verbose form of "BOX" should be "boxcar" rather than "rectangle" - this will be less confusing.